test/Microsoft.ML.Tests/DatabaseLoaderTests.cs

```diff
@@ -225,6 +239,22 @@ private DatabaseSource GetIrisDatabaseSource(string command)
                     String.Format(command, TestDatasets.irisDbSQLite.trainFilename));
         }
 
+        private DatabaseSource GetIrisDatabaseSourceWithTimeOut(string command, int commandTimeOut)
```

Comment:
There is code repetition here we can avoid. Perhaps it'd be better if this `GetIrisDatabaseSourceWithTimeOut` were combined with `GetIrisDatabaseSource`, where `commandTimeOut` is an optional input, and where time out would be added in `DatabaseSource`'s constructor if a `commandTimeOut` is actually given.

Reply:
Thanks for the comment! It's a good idea but considering that DatabaseSource has two ctors(overloading, avoid breaking API change), if we combine GetIrisDatabaseSourceWithTimeOut() with  GetIrisDatabaseSource() we still need to separately call that two ctors.

Reply:
And if the constructor without a timeout uses a default timeout, is it possible to merge the two implementations?

---
In reply to: [451054157](https://github.com/dotnet/machinelearning/pull/5288#discussion_r451054157) [](ancestors = 451054157)

Reply:
Then yes we can merge them and exercise the new added ctor only